api/tx_args.go

```diff
@@ -146,7 +142,27 @@ func (args *SendTxArgs) setDefaults(ctx context.Context, b Backend) error {
 		nonce := b.GetPoolNonce(ctx, args.From)
 		args.AccountNonce = (*hexutil.Uint64)(&nonce)
 	}
-
+	// Estimate the gas usage if necessary.
+	if args.GasLimit == nil {
```

Comment:
As @aidan-kwon said, present estimation logic can calculate accurate gas for only legacyTx. So how about about limiting this calculation to legacy transactions only?
```suggestion
	if args.GasLimit == nil && *args.TypeInt == types.TxTypeLegacyTransaction{
```

Reply:
@KimKyungup If it can be only applied to the legacy transactions, let's leave them as-is and change them all together when we are able to calculate the gas for all types of transactions? cc @aidan-kwon

Reply:
Ok. I got it.